addons/remember-collapsed-categories/userscript.js

```diff
@@ -0,0 +1,56 @@
+export default async function ({ console, addon }) {
+  while (true) {
+    const categoryBody = await addon.tab.waitForElement('[id^="category_body_"]', {
+      markAsSeen: true,
+    });
+    console.log(`Found category ${categoryBody.id}`);
+
+    const categoryCollapse = categoryBody.querySelector(".toggle");
+    categoryCollapse.addEventListener("click", (event) => {
+      event.preventDefault();
+      const categoryHead = event.target.parentElement;
+      categoryHead.classList.toggle("sa-collapsed");
+      if (categoryHead.classList.contains("sa-collapsed")) {
+        console.log(`${categoryHead.id.split("_")[2]} is being opened`);
+        return;
+      } else {
+        extendCategoryCookieExistenceLength(categoryNumber);
+      }
+    });
+
+    const categoryNumber = categoryBody.id.split("_").pop();
+    if (!shouldCollapseCategory(categoryNumber)) {
+      removeCategoryCookie(categoryNumber);
```

Comment:
Will this ever delete an existing cookie? I can't think of any situations where this function call would do anything.

Reply:
Normally Scratch sets the cookie to just an empty value, it doesn't delete it outright. Although it set the cookie's `max-length` to the session.

But I'm not sure about your question, are you asking if it could potentially delete the `scratchsession` cookie? If that is your question: the answer is no, it takes in a number as the input, this is used to delete the cookie that uses the naming scheme of `category_body_[number]`, in other words, the category cookies.

Reply:
> are you asking if it could potentially delete the `scratchsession` cookie?

I was not referring to that.